### ISSUE TITLE
Freemoji emote lookup in chat not showing external emotes

### ISSUE THREAD

**Issue by pkubatrh** [QbDesu/BetterDiscordAddons#13]
**Describe the bug**
While using the emote lookup dialog that gets shown after writing `:emotename...` only that server's emotes are shown.

I did not notice this happening after some specific update, but I might have missed it. Looking at the code trying to find what might be the cause of this I noticed this line of code: https://github.com/QbDesu/BetterDiscordAddons/blob/potato/Plugins/Freemoji/Freemoji.plugin.js#L300
```
Patcher.after(EmojiFilter, 'getEmojiUnavailableReason', (_, [{intention, bypassPatch}], ret) => {
                        if (intention!==EmojiIntention.CHAT || bypassPatch || !this.settings.external) return;
         ->             return ret===EmojiDisabledReasons.DISALLOW_EXTERNAL ? null : ret;
                    });
```

If I understand correctly (I might certainly be wrong since it has been years since I last loked at javascript code) this line checks if the value returned by `getEmojiUnavailableReason` is `EmojiDisabledReasons.DISALLOW_EXTERNAL` and if it is, then it returns `null` instead. Returns `ret` (unchanged return value) otherwise.
Should it not be the other way around? I saw the plugin checking for `EmojiDisabledReasons.DISALLOW_EXTERNAL` elsewhere, so it does not make much sense to ignore it if we want to check it later on.

**Comment by FlurryShy** [comment]
Noticed the same after todays update.
as a workaround your can press [ctrl] + [e] to open the emoji picker and continue typing.
has the added benefit that it would show all that match and not just the limited list you'd get with the :emotename 😊

**Comment by pkubatrh** [comment]
Easy workaround for me was to just rewrite the return from the function I pointed out above so that it always returns `null`. Seems to work fine for me as I ignore server-based external emote settings anyway. Just an idea for other folks who have the same use case until a proper update is available.

**Comment by Achenar459** [comment]
Can you share the rewrite for those of us who are code-illiterate? 😊

**Comment by ItsMeNoodle** [comment]
> Can you share the rewrite for those of us who are code-illiterate? 😊

`{
                        if (intention!==EmojiIntention.CHAT || bypassPatch || !this.settings.external) return;
                        return ret===EmojiDisabledReasons.DISALLOW_EXTERNAL ? null : ret;
                    }`
                  
here :)

**Comment by ghost** [comment]
> Can you share the rewrite for those of us who are code-illiterate? 😊

here :)
`Patcher.after(EmojiFilter, 'getEmojiUnavailableReason', (_, [{intention, bypassPatch}], ret) => null);`

**Comment by QbDesu** [comment]
Should be fixed in the 1.6.0 update. At least it does for me. The fix provided above while probably working would would have broken the ability to use external emoji when you don't have external emoji permissions.
If you still run into trouble be sure to let me know.
